pynbody/halo/ahf.py

```diff
@@ -317,7 +317,7 @@ def _load_ahf_particle_block(self, f, nparts=None):
             if self._use_iord:
                 data = self._iord_to_fpos[data]
             else:
-                if type(self.base) is not snapshot.nchilada.NchiladaSnap:
+                if not isinstance(self.base, snapshot.nchilada.NchiladaSnap):
```

Comment:
There are multiple places where the syntax `if type(foo) is Bar` was used, where it could be `if isinstance(foo, Bar)`.
The former does not work with subclasses while the latter do. I have replaced the ones that appeared to fall in the second category.